File: algorithms/sort/quickSort/QuickSort.java
import java.util.Random;

class QuickSort{

	public static void quickSort(int arr[], int low, int high){

		if(high-low > 1){
			int pivot = high-1;
			int cmp = low-1;
			int temp;
			for(int i=low; i<pivot; i++){
				if(arr[i] < arr[pivot] ){
					cmp++;
					temp = arr[cmp];
					arr[cmp] = arr[i];
					arr[i] = temp;
				}
			}
			cmp++;
			temp = arr[pivot];
			arr[pivot] = arr[cmp];
			arr[cmp] = temp;
			quickSort(arr, low, cmp);
			quickSort(arr, cmp+1, high);
		}
	}

	public static void quickSort(int arr[]){
		quickSort(arr, 0, arr.length);
	}

	static void printArray(int arr[]){
		for(int x : arr)
			System.out.print(x + ", ");
	}

	public static void main(String args[]){

		Random r = new Random();
		int arr[] = new int[10];

		for(int i=0; i<10; i++){
			arr[i] = r.nextInt(200);
		}

		System.out.print("Orignal array : ");
		printArray(arr);

		quickSort(arr);

		System.out.print("\n Sorted array : ");
		printArray(arr);

		System.out.println();
	}
}

File: algorithms/sort/quickSort/Quick_Sort.py
from random import randint

def Partition(A, low, high):

    Pivot = A[high-1]
    i = low-1

    for j in range(low, high-1):
        if(A[j] < Pivot):
            i+=1
            A[i], A[j] = A[j], A[i]
    A[i+1], A[high-1] = A[high-1], A[i+1]
    return i+1

def QuickSort(A, low, high):
    if(high-low > 1):
        mid = Partition(A, low, high)
        QuickSort(A, low, mid)
        QuickSort(A, mid+1, high)

A = [randint(-200,200) for i in range(10)]
print("Orignal array = ", *A)
QuickSort(A, 0, len(A))
print(" Sorted array = ", *A)
